controllers/grid/articleGalleys/form/ArticleGalleyForm.php

```diff
@@ -25,34 +25,42 @@
 
 class ArticleGalleyForm extends Form
 {
-    /** @var Submission */
+    /** @var \Submission */
     public $_submission = null;
 
-    /** @var Publication */
+    /** @var \Publication */
     public $_publication = null;
 
     /** @var Galley current galley */
     public $_articleGalley = null;
 
+    public bool $_isEditable = true;
+
     /**
      * Constructor.
      *
-     * @param Submission $submission
-     * @param Publication $publication
+     * @param \Submission $submission
+     * @param \Publication $publication
      * @param Galley $articleGalley (optional)
+     * @param bool $isEditable (optional, default = true)
      */
-    public function __construct($request, $submission, $publication, $articleGalley = null)
+    public function __construct($request, $submission, $publication, $articleGalley = null, bool $isEditable = true)
     {
         parent::__construct('controllers/grid/articleGalleys/form/articleGalleyForm.tpl');
         $this->_submission = $submission;
         $this->_publication = $publication;
         $this->_articleGalley = $articleGalley;
+        $this->_isEditable = $isEditable;
 
         $this->addCheck(new \PKP\form\validation\FormValidator($this, 'label', 'required', 'editor.issues.galleyLabelRequired'));
         $this->addCheck(new \PKP\form\validation\FormValidatorRegExp($this, 'urlPath', 'optional', 'validator.alpha_dash_period', '/^[a-zA-Z0-9]+([\\.\\-_][a-zA-Z0-9]+)*$/'));
         $this->addCheck(new \PKP\form\validation\FormValidatorPost($this));
         $this->addCheck(new \PKP\form\validation\FormValidatorCSRF($this));
 
+        $this->addCheck(new \PKP\form\validation\FormValidatorCustom($this, '', 'optional', 'form.notEditable', function ($isEditable) {
+            return $isEditable;
+        }));
```

Comment:
I haven't checked but are you sure that this works? The `FormValidatorCustom` will pass a value into the callback function, but I don't think this value will be `$isEditable`. Did you mean to do the following?

```php
function() use ($isEditable) {
```

Reply:
In fact this is my first try to add the validation that we finally added in the `validate` function. I am removing this check.